interfaces/cos_agent/v0/README.md

```diff
@@ -0,0 +1,115 @@
+# `cos_agent`
+
+## Usage
+
+This relation interface describes the expected behavior of any charm claiming to be able to provide or require the `cos_agent` interface.
+
+In most cases, this will be accomplished using the [`cos_agent` library](https://charmhub.io/grafana-agent/libraries/cos_agent), although charm developers are free to provide alternative libraries as long as they fulfill the behavioral and schematic requirements described in this document.
+
+## Direction
+
+```mermaid
+flowchart TD
+    Provider -- "relation_name
+    metrics_endpoints
+    metrics_rules_dir
+    logs_rules_dir
+    recursive_rules_dir
+    log_slots
+    dashboard_dirs
+    refresh_events" --> Requirer
+    Requirer -- refresh_events --> Provider
+```
+
+As all Juju relations, the `cos_agent` interface consists of a provider and a requirer. In this case the `Provider` side of the relationship may provide telemetry settings, if not, the required side will use default values.
+
+## Behavior
+
+Since this interface is meant to be simple and lightweight each telemetry parameter in the `Provider` side is optional, if some of these are not provided, dafault values will be used.
+
+### Provider
+
+- Is expected to be able to provide metrics endpoints to be scraped.
+- Is expected to be able to provide a directory containing Prometheus style metrics rules files.
+- Is expected to be able to provide a directory containing Loki style rules files.
+- Is expected to be able to inform if metrics and logs rules directories should be scanned recursively.
+- Is expected to be able to provide snap slots to connect to for scraping logs.
+- Is expected to be able to provide a list of directories containing Grafana dashboard files.
+- Is expected to be able to provide a list of events on which to refresh relation data.
```

Comment:
**question:** Are those library specs or interface specs?

Reply:
I think are both...

Reply:
Shouldn't this be strictly interface specs?
I.e. the contents of the `*.json` has nothing to do with
- a directory containing Prometheus style metrics rules files
- inform if metrics and logs rules directories should be scanned recursively
- etc.

but I guess I'm missing something..?